src/plugins/discover/public/application/main/services/discover_saved_search_container.ts

```diff
@@ -169,6 +175,26 @@ export function getSavedSearchContainer({
     }
     return { id };
   };
+
+  const assignNextSavedSearch = ({ nextSavedSearch }: { nextSavedSearch: SavedSearch }) => {
+    const hasChanged = !isEqualSavedSearch(savedSearchInitial$.getValue(), nextSavedSearch);
+    hasChanged$.next(hasChanged);
+    savedSearchCurrent$.next(nextSavedSearch);
+  };
+
+  const updateWithFilterManagerFilters = ({ savedSearch }: { savedSearch: SavedSearch }) => {
+    const nextSavedSearch: SavedSearch = {
+      ...savedSearch,
+    };
```

Comment:
we could consider to do it like the update function, getState() always returns the most recent SavedSearch state, so it doesn't need to be passed in from consumers ...
```suggestion
  const updateWithFilterManagerFilters = () => {
    const nextSavedSearch: SavedSearch = {
      ...getState(),
    };
```